docs/proposals/devfile_and_default_ports_detection.md

```diff
@@ -0,0 +1,91 @@
+# Devfile port detection & default ports
+
+## Related Github Issues
+
+- https://github.com/devfile/api/issues/1154
+
+## Background
+In order to perform port detection component wise, `alizer` is performing different strategies of port detection per component. Currently, we support 3 different strategies:
+
+* Dockerfile Port Detection
+* Docker Compose Port Detection
+* Source Code Port Detection
+
+At the moment, `alizer` doesn't support port detection from `devfile.y(a)ml` files. This means that all ports defined inside a file like this will not be included in alizer's response. _As a result, the first goal of this proposal is to include the devfiles as part of the port detection algorithm_, exactly like `dockerfile` and `compose`.
+
+On the other hand, right now `alizer` is not returning any default ports of detected frameworks. As a result, for every component using a framework that has default ports, alizer will not include them in its response even if we haven't detected any ports from our strategies (ports list is empty). That said, as a second goal of this proposal, we would like to add a new feature in alizer that will include default ports in alizer's response.
+
+## Devfile Port Detection
+For devfile port detection we should follow the same flow with dockerfile:
+
+1. Create a function that will:
+    * look for a devfile inside a given component (The `utils.GetLocations` function should be updated to support devfiles too).
```

Comment:
Would you like to elaborate more on this item? Especially how devfile files will be detected and in which order. I am thinking in particular about https://github.com/devfile/api/issues/1265 and https://github.com/devfile/api/issues/1035. Seeing Alizer as a tool of the Devfile library, I guess it makes sense to clarify the behavior for example in case there are multiple `{.,}devfile.y{,a}ml` in the component directory. Or do you plan to return all ports read from all valid default devfiles?

Reply:
Also, it might be useful to have a CLI flag to specify the devfile path..

Reply:
My suggestion is to follow the same flow as the `d(D)ockerfile`/`c(C)ontainerfile` for the docker port detection. More detailed, Alizer is looking on the root level and one level down inside each component. If it finds a file inside the root level it will return this as the detected one. For example, let's assume we have the following components:

```
componentA/
-- Dockerfile
-- main.py
-- requirements.txt

componentB/
-- docker/
-- -- Dockerfile
-- main.py
-- requirements.txt

componentC/
-- docker/
-- -- docker-internal/
-- -- -- Dockerfile.other
-- main.py
-- requirements.txt

componentD/
-- docker/
-- -- Dockerfile
-- Dockerfile
-- main.py
-- requirements.txt
```

For `componentA` and `componentB` we will be able to detect dockerfiles. For `componentC` we won't. Finally for `componentD` we will only detect the root level `dockerfile`. The same I'm suggesting to apply for `devfile`.

Reply:
Ok, that makes sense.